### ISSUE TITLE
Can't create queries within coroutine

### ISSUE THREAD

**Issue by ARitz-Cracker** [syl0r/MySQLOO#9]
```
require( "mysqloo" )
local db = mysqloo.connect( "192.168.0.4", "tester" , "tester123" , "test_db", 3306)
print(db)
function db:onConnected()
    print("Connected!")
    local t = coroutine.create( function()
        local q = db:query( "SELECT 1+1;" )
        function q:onSuccess( data )
            PrintTable(data)
        end
        function q:onError( err, sqlq )
            print("failed to query "..err)
        end
        q:start()
        print(q)
    end)
    
    coroutine.resume( t )
end
function db:onConnectionFailed(err)
    print("failed to connect "..err)
end
print(db.connect)
db:connect()
```
results in
```
fucking_bullshit.lua:18: bad argument #1 to 'resume' (table expected, got thread)
```

**Comment by ARitz-Cracker** [comment]
It appears that part of the solution is to replace all instances of `int funcName( lua_State* state ) to `LUA_FUNCTION( funcName )`, [as seen here](https://github.com/Facepunch/gmod-module-base/blob/development/include/GarrysMod/Lua/Interface.h#L42) since the extra code in the macro is what fixes the issue.

But this is your module, so I don't want to re-code it myself.

**Comment by ARitz-Cracker** [comment]
The awesome MetaMan has submitted a PR to fix this #10
